### ISSUE TITLE
Track/record features for release notes

### ISSUE THREAD

**Issue by david-mcmahon** [kubernetes/release#122]
Moved from https://github.com/kubernetes/kubernetes/issues/23070

Currently, PRs having release-note label is automatically included as part of the release notes. This process does not work for big features though.
We need to track issues that represent features as well and include a gist from those issues in the release notes.

cc @vishh @spiffxp @pwittrock @erictune

**Comment by david-mcmahon** [comment]
From email to kubernetes-dev:

One thing that was clear is that most people didn't know what should go into the release notes. In a number of cases, I had to ask specific people to add specific items to specific sections. The only way I could do that is I knew what was in the release and how the release notes should be organized.

We need a template for release notes, guidelines for what belongs in release notes, and an example set of release notes (1.2.0).

We also need to capture items that belong in release notes before the PRs merge.

David M. has a proposal to automate release-note collection. We need to extend it with the categories we've created and require every PR assign a category (including something like release-note/none-minor-fix) before being merged by the submit queue. PRs that belong in the release notes will need to have extractable release note text in their PR descriptions.

Writing release notes should be added to the documented developer workflow.
